resource/graph_node.go

```diff
@@ -252,6 +265,20 @@ func (w *GraphNode) setDependenciesResolved() {
 	w.needsDependencyResolution = false
 }
 
+// IncrementTimesReconfigured increments the number of times the resource has been
+// reconfigured by 1. Value resetting handled in other methods situationally.
+func (w *GraphNode) IncrementTimesReconfigured() {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	w.timesReconfigured++
+}
+
+// resetTimesReconfigured is a private method (no mutex!!!) that sets timesReconfigured
+// back to 0.
+func (w *GraphNode) resetTimesReconfigured() {
+	w.timesReconfigured = 0
```

Comment:
need lock

Reply:
Wouldn't that cause deadlock since `resetTimesReconfigured` is only called within other methods that acquire the lock

Reply:
If you go with my suggestion of using an atomic, none of these methods should require locks.